File: dart/2168.dart
import 'dart:io';

void main() {
  int n = int.parse(stdin.readLineSync()!);
  List<List<String>> l = [];
  String output = "";

  for (int i = 0; i < n+1; i++) {
    l.add(stdin.readLineSync()!.split(" "));
  }

  for (int i = 0; i < n; i++) {
    for (int j = 0; j < n; j++) {
      int x = int.parse(l[i][j]) + int.parse(l[i][j+1]) + int.parse(l[i+1][j]) + int.parse(l[i+1][j+1]);
      if (x > 1) {
        output += "S";
      } else {
        output += "U";
      }
    }

    if (i < n-1) {
      output += "\n";
    }
  }

  print(output);
}

File: python/2168.py
a = int(input())
i = 0
j = 0
x = 0
l = []

while i < a+1:
    l.append(input().split())
    i += 1

i = 0

while i < a:
    while j < a:
        x = int(l[i][j]) + int(l[i][j+1]) + int(l[i+1][j]) + int(l[i+1][j+1])
        if x > 1:
            print("S", end="")
        else:
            print("U", end="")
        if j == a-1:
            print()
        x = 0
        j += 1
    j = 0
    i += 1